### PR TITLE
refactor(core): move 'watch:compile' from make to buck

### PR DESCRIPTION
**Pull request by vindard**
## Description

This change would mean that to watch-compile we would now do:
```bash
$ buck2 run //core/api:watch-compile
```

cc @nicolasburtey @dolcalmi

### CODE REVIEW

**Review by bodymindarts — DISMISSED** [review]

**Comment by bodymindarts** [comment]
Approved but lets see if it works for @dolcalmi

File: bats/helpers/subscriber/output.txt
Comment (dolcalmi) [inline bats/helpers/subscriber/output.txt:1]: what is the purpose of this file?

File: core/api/output.txt
Comment (dolcalmi) [inline core/api/output.txt:1]: same here

File: test_script.sh
Comment (dolcalmi) [inline test_script.sh:1]: where is this used?

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: test_script.sh
Comment (vindard) [reply test_script.sh:1]: Ah sorry good catch, these got accidentally committed

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**